src/Sentry.AspNetCore/SentryStartupFilter.cs

```diff
@@ -11,10 +16,19 @@ public class SentryStartupFilter : IStartupFilter
     /// <summary>
     /// Adds Sentry to the pipeline.
     /// </summary>
-    public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next) => e =>
+    public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next) => app =>
     {
-        e.UseSentry();
+        // If we are capturing request bodies and the user has configured request body decompression, we need to
+        // ensure that the RequestDecompression middleware gets called before Sentry's middleware.
+        var options = app.ApplicationServices.GetService<IOptions<SentryAspNetCoreOptions>>();
+        if (options?.Value is { } o && o.MaxRequestBodySize != RequestSize.None
+            && app.ApplicationServices.GetService<IRequestDecompressionProvider>() is not null)
+        {
+            app.UseMiddleware<RequestDecompressionMiddleware>();
+        }
 
-        next(e);
+        app.UseSentry();
+
+        next(app);
```

Comment:
question: Double-decompression?

If the consuming App-Code has both `builder.Services.AddRequestDecompression()` and also `app.UseRequestDecompression()`, would the Stream now be decompressed twice in the pipline (via both Sentry's and AspNetCore's `RequestDecompressionMiddleware`).

Or is this prevented by
https://github.com/dotnet/aspnetcore/blob/8883b98afedee35e8360db3f24ce8cd2e17217ad/src/Middleware/RequestDecompression/src/DefaultRequestDecompressionProvider.cs#L52?

Reply:
If the body is uncompressed (which it will be after we run our custom ) then `_provider.GetDecompressionStream` returns null (see [docs](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.requestdecompression.irequestdecompressionprovider.getdecompressionstream?view=aspnetcore-9.0#returns)). 

So the second middleware [does nothing](https://github.com/dotnet/aspnetcore/blob/9e69f19a5c7efa19577627a62c029ff387810d50/src/Middleware/RequestDecompression/src/RequestDecompressionMiddleware.cs#L47-L51) basically.